echopype/tests/echodata/test_echodata_structure.py

```diff
@@ -0,0 +1,271 @@
+
+    # make sure that the dimensions and attributes
+    # are the same for the variable
+    assert ed_var.dims == tree_var.dims
+    assert ed_var.attrs == tree_var.attrs
+
+    # make sure that the data types are correct too
+    assert isinstance(ed_var.values, type(tree_var.values))
+
+    # drop variables so we can check that datasets are identical
+    ed[grp_path] = ed[grp_path].drop(var)
+    tree[grp_path].ds = tree[grp_path].ds.drop(var)
+
+
+def _check_and_drop_attr(ed, tree, grp_path, attr, typ):
+    """
+    This function performs minimal checks of
+    an attribute contained both in an EchoData object
+    and a Datatree group. This function only works for
+    a group's attribute, it cannot work on variable
+    attributes. It ensures that the attribute exists
+    and that it has the expected data type. Once
+    the checks have passed, it then drops the
+    attribute from both the EchoData object and the
+    Datatree.
+
+    Parameters
+    ----------
+    ed : EchoData
+        EchoData object that contains the attribute
+        to check and drop.
+    tree : Datatree
+        Datatree object that contains the attribute
+        to check and drop.
+    grp_path : str
+        The path to the group that the attribute  is in.
+    attr : str
+        The attribute to be checked and dropped.
+    typ : type
+        The expected data type of the attribute.
+
+    Notes
+    -----
+    The Datatree object is created from an EchoData
+    object written to a netcdf file.
+    """
+
+    # make sure that the attribute exists
+    assert attr in ed[grp_path].attrs.keys()
+    assert attr in tree[grp_path].ds.attrs.keys()
+
+    # make sure that the value of the attribute is the right type
+    assert isinstance(ed[grp_path].attrs[attr], typ)
+    assert isinstance(tree[grp_path].ds.attrs[attr], typ)
+
+    # drop the attribute so we can directly compare datasets
+    del ed[grp_path].attrs[attr]
+    del tree[grp_path].ds.attrs[attr]
+
+
+def compare_ed_against_tree(ed, tree):
+    """
+    This function compares the Datasets
+    of ed against tree and makes sure they
+    are identical.
+
+    Parameters
+    ----------
+    ed : EchoData
+        EchoData object
+    tree : Datatree
+        Datatree object
+
+    Notes
+    -----
+    The Datatree object is created from an EchoData
+    object written to a netcdf file.
+    """
+
+    for grp_path in ed.group_paths:
+        if grp_path == "Top-level":
+            assert tree.ds.identical(ed[grp_path])
+        else:
+            assert tree[grp_path].ds.identical(ed[grp_path])
+
+
+def test_v05x_v06x_conversion_structure(azfp_path, ek60_path, ek80_path):
+    """
+    Tests that version 0.5.x echopype files
+    have been correctly converted to the
+    0.6.x structure.
+    """
+
+    converted_raw_paths_v06x = [ek60_path / "ek60-Summer2017-D20170615-T190214-ep-v06x.nc",
+                                ek80_path / "ek80-Summer2018--D20180905-T033113-ep-v06x.nc",
```

Comment:
We probably should identify a couple of contrasting EK80 raw files to test, rather than just one, to catch EK80 variability; eg, one having both complex and power data, vs just power.

Reply:
I agree. They are explicitly listed in the tests in `tests/calibrate/test_calibrate.py`

Reply:
I have added extra EK80 raw files to test in my most recent push